base/reduce.jl

```diff
@@ -260,9 +260,7 @@ mapreduce_empty(f::typeof(abs),  ::typeof(max), T) = abs(zero(T))
 mapreduce_empty(f::typeof(abs2), ::typeof(max), T) = abs2(zero(T))
 
 mapreduce_empty_iter(f, op, itr, ::HasEltype) = mapreduce_empty(f, op, eltype(itr))
-mapreduce_empty_iter(f, op::typeof(&), itr, ::EltypeUnknown) = true
-mapreduce_empty_iter(f, op::typeof(|), itr, ::EltypeUnknown) = false
-mapreduce_empty_iter(f, op, itr, ::EltypeUnknown) = _empty_reduce_error()
+mapreduce_empty_iter(f, op, itr, ::EltypeUnknown) = mapreduce_empty(f, op, Any)
```

Comment:
Can we just delete the type constraint on Line 262, for `HasEltpe`?
Since `eltype(::Any)=Any` anyway

Though this is also fine 🤷‍♂️

Reply:
I guess so. Are there any cases where `IteratorEltype(itr) == EltypeUnknown()` but `eltype(itr)` would throw an error?

Reply:
I'd rather keep it, in hope of being able to remove the eltype fallback one day.

Reply:
Okay, kept.